diff --git a/includes/media_youtube.formatters.inc b/includes/media_youtube.formatters.inc
index a0ad95e..56beebb 100644
--- a/includes/media_youtube.formatters.inc
+++ b/includes/media_youtube.formatters.inc
@@ -270,67 +270,3 @@ function media_youtube_file_formatter_image_settings($form, &$form_state, $setti
   );
   return $element;
 }
-
-/**
- * Implements hook_file_default_displays_alter().
- */
-function media_youtube_file_default_displays_alter(&$file_displays) {
-  // Video previews should be displayed using a large filetype icon.
-  $file_display = new stdClass();
-  $file_display->api_version = 1;
-  $file_display->name = 'video__default__media_youtube_video';
-  $file_display->weight = 0;
-  $file_display->status = TRUE;
-  $file_display->settings = array(
-    'width' => '640',
-    'height' => '390',
-    'theme' => 'dark',
-    'color' => 'red',
-    'autohide' => '2',
-    'autoplay' => 0,
-    'loop' => 0,
-    'showinfo' => 1,
-    'modestbranding' => 0,
-    'rel' => 1,
-    'nocookie' => 0,
-    'protocol_specify' => 0,
-    'protocol' => 'https:',
-    'enablejsapi' => 0,
-    'origin' => '',
-  );
-  $file_displays['video__default__media_youtube_video'] = $file_display;
-
-  $file_display = new stdClass();
-  $file_display->api_version = 1;
-  $file_display->name = 'video__preview__media_youtube_image';
-  $file_display->weight = 0;
-  $file_display->status = TRUE;
-  $file_display->settings = array(
-    'image_style' => 'media_thumbnail',
-  );
-  $file_displays['video__preview__media_youtube_image'] = $file_display;
-
-  $file_display = new stdClass();
-  $file_display->api_version = 1;
-  $file_display->name = 'video__teaser__media_youtube_video';
-  $file_display->weight = 0;
-  $file_display->status = TRUE;
-  $file_display->settings = array(
-    'width' => '560',
-    'height' => '340',
-    'theme' => 'dark',
-    'color' => 'red',
-    'autohide' => '2',
-    'autoplay' => 0,
-    'loop' => 0,
-    'showinfo' => 1,
-    'modestbranding' => 0,
-    'rel' => 1,
-    'nocookie' => 0,
-    'protocol_specify' => 0,
-    'protocol' => 'https:',
-    'enablejsapi' => 0,
-    'origin' => '',
-  );
-  $file_displays['video__teaser__media_youtube_video'] = $file_display;
-}
diff --git a/media_youtube.install b/media_youtube.install
index ed8acd1..66ccc64 100644
--- a/media_youtube.install
+++ b/media_youtube.install
@@ -6,6 +6,88 @@
  */
 
 /**
+ * Implements hook_install().
+ */
+function media_youtube_install() {
+  _media_youtube_import_file_displays();
+}
+
+/**
+ * Import default file displays.
+ */
+function _media_youtube_import_file_displays() {
+  // Create initial display settings.
+  module_load_include('inc', 'file_entity', 'file_entity.file_api');
+
+  $existing_display = file_displays_load('video', 'default');
+  if (empty($existing_display)) {
+    // Video previews should be displayed using a large filetype icon.
+    $display = array(
+      'api_version' => 1,
+      'name' => 'video__default__media_youtube_video',
+      'weight' => 0,
+      'status' => 1,
+      'settings' => array(
+        'width' => '640',
+        'height' => '390',
+        'theme' => 'dark',
+        'color' => 'red',
+        'autohide' => '2',
+        'autoplay' => 0,
+        'loop' => 0,
+        'showinfo' => 1,
+        'modestbranding' => 0,
+        'rel' => 1,
+        'nocookie' => 0,
+        'protocol_specify' => 0,
+        'protocol' => 'https:',
+        'enablejsapi' => 0,
+        'origin' => '',
+      ),
+    );
+    file_display_save((object) $display);
+
+    $display = array(
+      'api_version' => 1,
+      'name' => 'video__preview__media_youtube_image',
+      'weight' => 0,
+      'status' => 1,
+      'settings' => array(
+        'image_style' => 'media_thumbnail',
+      ),
+    );
+    file_display_save((object) $display);
+
+    $display = array(
+      'api_version' => 1,
+      'name' => 'video__teaser__media_youtube_video',
+      'weight' => 0,
+      'status' => 1,
+      'settings' => array(
+        'width' => '560',
+        'height' => '340',
+        'theme' => 'dark',
+        'color' => 'red',
+        'autohide' => '2',
+        'autoplay' => 0,
+        'loop' => 0,
+        'showinfo' => 1,
+        'modestbranding' => 0,
+        'rel' => 1,
+        'nocookie' => 0,
+        'protocol_specify' => 0,
+        'protocol' => 'https:',
+        'enablejsapi' => 0,
+        'origin' => '',
+      ),
+    );
+    file_display_save((object) $display);
+
+  }
+
+}
+
+/**
  * Clear style and preset caches.
  */
 function media_youtube_update_7001() {
@@ -156,3 +238,11 @@ function media_youtube_update_7203() {
     variable_del("media_youtube__{$setting}");
   }
 }
+
+/**
+ * Move display defaults to the db instead of hook_file_default_displays_alter().
+ */
+function media_youtube_update_7204() {
+  _media_youtube_import_file_displays();
+  return t('The Media Youtube default file displays were moved to the database.');
+}
