diff --git a/imagecache_external.module b/imagecache_external.module
index 3622365..1ed2174 100755
--- a/imagecache_external.module
+++ b/imagecache_external.module
@@ -118,14 +118,15 @@ function theme_imagecache_external($variables) {
 function imagecache_external_generate_path($url, $preset) {
   // Create the extenal images directory and ensure it's writable.
   $hash       = md5($url);
+  $filename   = $hash . '.' . substr(strrchr($url,'.'),1);
   $scheme     = file_default_scheme();
   $directory  = $scheme . '://imagecache/externals';
   if(file_prepare_directory($directory, FILE_CREATE_DIRECTORY | FILE_MODIFY_PERMISSIONS)) {
-    if (file_exists($directory. '/' . $hash)) {
-      return $directory. '/' . $hash;
+    if (file_exists($directory. '/' . $filename)) {
+      return $directory. '/' . $filename;
     }
 
-    $file = imagecache_external_fetch($url, $directory . '/' . $hash);
+    $file = imagecache_external_fetch($url, $directory . '/' . $filename);
     if ($file) {
       return $file->uri;
     }
